storage/index.ts

```diff
@@ -48,33 +71,39 @@ export function getMetadata(ref: Reference): Observable<any> {
 }
 
 export function put(
-    ref: Reference,
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    data: any,
-    metadata?: UploadMetadata,
+  ref: Reference,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  data: any,
```

Comment:
Feel free to ignore this since it's off topic from the main purpose of this PR, but want to make this `Blob | Uint8Array | ArrayBuffer` to [match the source](https://github.com/firebase/firebase-js-sdk/blob/cc7207e25f09870c6c718b8e209e694661676d27/packages/storage-types/index.d.ts#L42)

Reply:
Absolutely, also a good time to clean up the types as we're breaking.

Reply:
I'll do that in #10